firmware/crop.c

```diff
@@ -0,0 +1,76 @@
+
+#include <generated/csr.h>
+#include <generated/mem.h>
+#include <hw/flags.h>
+#include <system.h>
+#include <time.h>
+
+#include "hdmi_in0.h"
+#include "hdmi_in1.h"
+#include "crop.h"
+#include "processor.h"
+#include "pattern.h"
+#include "stdio_wrap.h"
+
+static bool crop_stat = false;
+
+
+
+void crop_status(bool val)
+{
+	crop_stat = val;
+}
+
+void crop_service(fb_ptrdiff_t fb_offset,int top,int bottom,int left,int right)
+{
+	static int last_event;
+	static int counter;
+	static bool color_v;
+
+	if (crop_stat==1) {
+		crop_fill(color_v, fb_offset,top,bottom,left,right);
+
+	}
+}
+
+void crop_fill(bool color_v, fb_ptrdiff_t fb_offset,int top,int bottom,int left,int right)
+{
+	int addr, i, j;
+	unsigned int color;
+	unsigned int *framebuffer = fb_ptrdiff_to_main_ram(fb_offset);
+
+//Bottom clipping
```

Comment:
Fix tab alignment. Comments should be in line with the code they are associated with